Facilitate fetching argument values for jobs

lars-petter-hauge opened the pull request:
**Issue**
Resolves #1039


**Approach**
The values for the arguments for a configured job was not exposed, added another function to facilitate retrieving the values from python.

Probably not handling the allocation correctly - must be checked.

Tests not added!

lars-petter-hauge committed:
Facilitate fetching argument values for jobs

The arguments are resolved at runtime when the jobs.json
file is printed. We include a function to extract the "what
will be" values for those jobs. This function will not be
exhaustive, as ERT provides some of the argument values, such
as ITER and IENS.

ManInFez commented on lib/enkf/model_config.cpp:
```diff
@@ -506,6 +508,7 @@ void model_config_init(model_config_type * model_config ,
 
     model_config->obs_config_file = util_alloc_string_copy(obs_config_file);
   }
+  // Only subst_list_free for non const. Free const? (define_list);
```
config should be responsible for cleaning up define_list

lars-petter-hauge replied:
As this is the initiation of the `ModelConfig` it _should_ free it? We are doing a deep copy of the list within `ext_job_set_define_args`, and as such the `ext_job` will not need it.

ManInFez replied:
The define_list is owned by the config_content object. I might be passed on to some other config object at a later stage.

ManInFez commented on python/res/job_queue/ext_job.py:
```diff
@@ -202,6 +203,9 @@ def get_license_path(self):
         return self._get_arglist()
 
+    def get_argvalues(self):
+        return self._get_argvalues()
```
should we make this a list and not have a stringList moving around in the python layer?

lars-petter-hauge replied:
I would second that, I did it to be consistent with `get_arglist`, but perhaps I can change both instead?

ManInFez replied:
I would say change both :+1:

lars-petter-hauge replied:
I just wrapped it in `list` (and added a small comment as to why..)

ManInFez commented on lib/job_queue/ext_job.cpp:
```diff
@@ -1107,6 +1113,21 @@ const stringlist_type * ext_job_get_arglist( const ext_job_type * ext_job ) {
     return ext_job->argv;
 }
 
+const stringlist_type * ext_job_get_argvalues( const ext_job_type * ext_job ) {
```
would it not make sense to create this when setting the arguments and not every time when extracting?

lars-petter-hauge replied:
It would require the `set_private_args` to be called before the `set_defined_args`, so I'm not sure.. We could do some checking if the `arg_values` have been resolved, it would add more state to the structure though.

ManInFez replied:
Is there some configuration where set_private_args and set_defined_args are called on their own?

lars-petter-hauge replied:
We do explicitly in the `forward_model_parse_job_deprecated_args` and `forward_model_parse_job_args`

ManInFez reviewed (commented):
Looks good! just a few things. Let me know if you want to discuss :)

lars-petter-hauge committed:
Change return type of getter in ExtJob

We like to avoid passing the C type containers out into the
python interface, and rather create ordinary python list.

ManInFez reviewed (approved):
Approved after black!
